zerver/webhooks/github/view.py

```diff
@@ -585,54 +610,54 @@ def get_tag_name_from_ref(ref_string: str) -> str:
     return re.sub(r"^refs/tags/", "", ref_string)
 
 
-def is_commit_push_event(payload: Dict[str, Any]) -> bool:
-    return bool(re.match(r"^refs/heads/", payload["ref"]))
+def is_commit_push_event(payload: WildValue) -> bool:
+    return bool(re.match(r"^refs/heads/", payload["ref"].tame(check_string)))
 
 
-def get_subject_based_on_type(payload: Dict[str, Any], event: str) -> str:
+def get_subject_based_on_type(payload: WildValue, event: str) -> str:
     if "pull_request" in event:
         return TOPIC_WITH_PR_OR_ISSUE_INFO_TEMPLATE.format(
             repo=get_repository_name(payload),
             type="PR",
-            id=payload["pull_request"]["number"],
-            title=payload["pull_request"]["title"],
+            id=payload["pull_request"]["number"].tame(check_int),
+            title=payload["pull_request"]["title"].tame(check_string),
         )
     elif event.startswith("issue"):
         return TOPIC_WITH_PR_OR_ISSUE_INFO_TEMPLATE.format(
             repo=get_repository_name(payload),
             type="issue",
-            id=payload["issue"]["number"],
-            title=payload["issue"]["title"],
+            id=payload["issue"]["number"].tame(check_int),
+            title=payload["issue"]["title"].tame(check_string),
         )
     elif event.startswith("deployment"):
         return "{} / Deployment on {}".format(
             get_repository_name(payload),
-            payload["deployment"]["environment"],
+            payload["deployment"]["environment"].tame(check_string),
         )
     elif event == "membership":
-        return "{} organization".format(payload["organization"]["login"])
+        return "{} organization".format(payload["organization"]["login"].tame(check_string))
     elif event == "team":
-        return "team {}".format(payload["team"]["name"])
+        return "team {}".format(payload["team"]["name"].tame(check_string))
     elif event == "push_commits":
         return TOPIC_WITH_BRANCH_TEMPLATE.format(
             repo=get_repository_name(payload),
-            branch=get_branch_name_from_ref(payload["ref"]),
+            branch=get_branch_name_from_ref(payload["ref"].tame(check_string)),
         )
     elif event == "gollum":
         return TOPIC_WITH_BRANCH_TEMPLATE.format(
             repo=get_repository_name(payload),
             branch="wiki pages",
         )
     elif event == "ping":
-        if payload.get("repository") is None:
+        if not payload.get("repository"):
```

Comment:
[Don’t use ambiguous truthiness tests](https://zulip.readthedocs.io/en/latest/testing/mypy.html#use-optional-and-none-correctly). Use `"repository" not in payload` to check for absence or `payload["repository"].value is None` to check for `null`.